### DISCUSSION TITLE
log right before server can start accepting

### DISCUSSION THREAD

**Discussion by prabirshrestha** [trillium-rs/trillium#27] [Q&A]
Is there an equivalent of this?

```rust
let mut app = tide::new();
let mut listener = app.bind(("0.0.0.0", "3000")).await?;
for info in listener.info().iter() {
    println!("Server listening on {}", info);
}
listener.accept().await?;
```

**Comment by jbr** [comment]
If it's just for logging, this happens by default if you enable a log backend. However, one of the things I've been thinking about is passing that information into each handler on the init hook. Is there something other than seeing the url that you're looking to do with this?

Given the [template application](https://github.com/trillium-rs/trillium-template), executing `RUST_LOG=info cargo run` results in:
![image](https://user-images.githubusercontent.com/13301/120940423-0daff800-c6d2-11eb-85f6-3e59942faee6.png)

**Reply by prabirshrestha** [reply]
I'm looking only for simple logging though I would want it to work even if I don't use `RUST_LOG=info`.

For example when my app starts I want to log the server started but other logs I don't want to show it yet.

**Reply by jbr** [reply]
If you don't add the Logger handler, the only info-level output will be the "listening on [::1]:8080" line. Does that work for your use case? Would you prefer it to go to stdout instead of `log`?

**Reply by prabirshrestha** [reply]
I would personally want at least this one to go to stdout instead of log but not sure if others will be ok with it.

**Reply by jbr** [reply]
Does this work for your use case?
```rust
    env_logger::Builder::new()
        .filter_module("trillium_server_common", log::LevelFilter::Info)
        .target(env_logger::Target::Stdout)
        .init();
    trillium_smol::run(…);
```

**Reply by jbr** [reply]
Update: When #47 is added, you'll be able to do

```rust
fn handler() -> impl Handler {
    (
        trillium::Init::new(|info| async move { println!("listening on {}", info) }),
        |conn: Conn| async move { conn.ok("ok") }
    )
}
```
